File: protocol/src/test/typescript/zfoo/packet/ObjectB.ts


class ObjectB {

    flag: boolean = false;

    static PROTOCOL_ID: number = 103;

    protocolId(): number {
        return ObjectB.PROTOCOL_ID;
    }

    static write(buffer: any, packet: ObjectB | null) {
        if (packet === null) {
            buffer.writeInt(0);
            return;
        }
        buffer.writeInt(-1);
        buffer.writeBoolean(packet.flag);
    }

    static read(buffer: any): ObjectB | null {
        const length = buffer.readInt();
        if (length === 0) {
            return null;
        }
        const beforeReadIndex = buffer.getReadOffset();
        const packet = new ObjectB();
        const result0 = buffer.readBoolean(); 
        packet.flag = result0;
        if (length > 0) {
            buffer.setReadOffset(beforeReadIndex + length);
        }
        return packet;
    }
}

export default ObjectB;

File: protocol/src/test/lua/LuaProtocol/Packet/ObjectB.lua
-- @author godotg

local ObjectB = {}

function ObjectB:new(flag)
    local obj = {
        flag = flag -- boolean
    }
    setmetatable(obj, self)
    self.__index = self
    return obj
end

function ObjectB:protocolId()
    return 103
end

function ObjectB:write(buffer, packet)
    if buffer:writePacketFlag(packet) then
        return
    end
    buffer:writeBoolean(packet.flag)
end

function ObjectB:read(buffer)
    if not(buffer:readBoolean()) then
        return nil
    end
    local packet = ObjectB:new()
    local result0 = buffer:readBoolean()
    packet.flag = result0
    return packet
end

return ObjectB
